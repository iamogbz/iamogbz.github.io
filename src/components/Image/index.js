import React from "react";
import PropTypes from "prop-types";

const imageAssetPath = "assets/images";

const Image = ({ name, size, type, className }) => (
    <img
        className={className}
        src={`/${imageAssetPath}/${name}.${type || Image.defaultProps.type}`}
        alt={name}
        width={size}
    />
);

Image.propTypes = {
    name: PropTypes.string.isRequired,
    className: PropTypes.string,
    size: PropTypes.string,
    type: PropTypes.string,
};

Image.defaultProps = {
    className: null,
    size: "initial",
    type: "png",
};

export default Image;
