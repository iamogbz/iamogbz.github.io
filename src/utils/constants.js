export const Colors = {
    ACTIVE: "#2386F1",
    LIGHT: "#e8e8e8",
    DARK: "#101010",
    NONE: "transparent",
};

export const Zindex = {
    TOP: 999,
    ABOVE: 1,
    NORMAL: 0,
    BELOW: -1,
    BOTTOM: -999,
};
