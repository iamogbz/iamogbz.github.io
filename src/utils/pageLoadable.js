import Loadable from "react-loadable";

import PageLoading from "components/PageLoading";

export default function pageLoadable(loader) {
    return Loadable({
        loader,
        loading: PageLoading,
    });
}
