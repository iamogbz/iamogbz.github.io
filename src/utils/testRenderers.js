import React from "react";
// import { mount, shallow } from "enzyme";
import { render } from "@testing-library/react";

const _render = (renderer, Component, props = {}) =>
    renderer(<Component {...props} />);

// export const mountRender = (...args) => _render(mount, ...args);
// export const shallowRender = (...args) => _render(shallow, ...args);
export const testRender = (...args) => _render(render, ...args).container.firstChild;
