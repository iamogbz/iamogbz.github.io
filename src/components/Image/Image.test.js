import { testRender } from "utils/testRenderers";
import Image from "./index";

describe("Image", () => {
    const render = (props = {}) => testRender(Image, props);

    it("renders with default type", () => {
        expect(render({ name: "test" })).toMatchSnapshot();
    });
});
