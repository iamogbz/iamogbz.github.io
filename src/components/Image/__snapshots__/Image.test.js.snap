// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders with default type 1`] = `
<img
  alt="test"
  src="/assets/images/test.png"
/>
`;
